File: C-in-class-projects/spring-2016/06/Navigation-From-Scratch/Navigation-From-Scratch/ViewController.m
//
//  ViewController.m
//  Navigation-From-Scratch
//
//  Created by Joseph Asante on 2/2/16.
//  Copyright © 2016 josephasante. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    UIBarButtonItem *compose = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCompose target:self action:@selector(composeAction:)];
    
    UINavigationItem *navItem = self.navigationItem;
    [navItem setRightBarButtonItem:compose];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)composeAction:(id)sender {
    // do stuff
    NSLog(@"composed message");
}

- (IBAction)transition:(id)sender {
    // transition
    UIViewController *secondViewController = [self.storyboard instantiateViewControllerWithIdentifier:@"secondView"];
    [self.navigationController pushViewController:secondViewController animated:YES];
}

@end

File: C-in-class-projects/fall-2016/08/SwiftProject/SwiftProject/ViewController.swift
//
//  ViewController.swift
//  SwiftProject
//
//  Created by Joseph Asante on 9/13/16.
//  Copyright © 2016 josephasante. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        print(String(funkyFunc("funky").int))
        var int:Int
        var str:String
        (str, int) = funkyFunc("funky")
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func funkyFunc(_ str:String) -> (str: String, int:Int) {
        return (str, 5)
    }

    
}

